File: vortexdbg-android/src/main/java/com/vortexdbg/linux/struct/RLimit64.java
package com.vortexdbg.linux.struct;

import com.vortexdbg.pointer.VortexdbgStructure;
import com.sun.jna.Pointer;

import java.util.Arrays;
import java.util.List;

public class RLimit64 extends VortexdbgStructure {

    public long rlim_cur;
    public long rlim_max;

    public RLimit64(Pointer p) {
        super(p);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("rlim_cur", "rlim_max");
    }

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/struct/RLimit64.kt
package com.vortexdbg.linux.struct

import com.vortexdbg.pointer.VortexdbgStructure
import com.sun.jna.Pointer

class RLimit64(p: Pointer?) : VortexdbgStructure(p) {

    @JvmField
    var rlim_cur: Long = 0
    @JvmField
    var rlim_max: Long = 0

    override fun getFieldOrder(): List<String> {
        return listOf("rlim_cur", "rlim_max")
    }

}
